### PR TITLE
Make sure all CDI crds have structural schemas

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
Ensure that CRDs related to CDI all have a compliant structural schema. Also ensure all fields have descriptions so they can be used with kubectl explain

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Enhancement: CDI CRDs are now all structural schemas.
```

### CODE REVIEW

File: pkg/operator/resources/cluster/cdiconfig.go [hunk 1]
```diff
@@ -28,6 +28,81 @@ func createCDIConfigCRD() *extv1beta1.CustomResourceDefinition {
 			},
 			Version: "v1alpha1",
 			Scope:   "Cluster",
+			Validation: &extv1beta1.CustomResourceValidation{
+				OpenAPIV3Schema: &extv1beta1.JSONSchemaProps{
+					Description: "CDIConfig is the configuration object for Containerized Data Importer",
+					Type:        "object",
+					Properties: map[string]extv1beta1.JSONSchemaProps{
+						"apiVersion": {
```
Comment (mhenriks) [inline pkg/operator/resources/cluster/cdiconfig.go:36]: According to this, can leave out apiVersion, kind, metadata:  https://kubernetes.io/blog/2019/06/20/crd-structural-schema/

Comment (awels) [reply pkg/operator/resources/cluster/cdiconfig.go]: Done

**Pull request merged**